[CS2113T-F14-1] Ha(ppy)Bit

DJ-Tan opened the pull request:
Ha(ppy)Bit is designed for students with preference for CLI over GUI and who aim for a fulfilling lifestyle (both physically and mentally), by cultivating good habits and replacing unhealthy ones.

adricpjw commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+The Sequence Diagram below illustrates the interactions within the `Ui` component when `run()` is called in the main
+class `HappyBit.java`
+
+![Ui Diagram](Diagram Images/UiDiagramCopy.png)
```
![image](https://user-images.githubusercontent.com/60023580/138998652-6e1b0b8e-b688-4848-9767-29919c79442e.png)
Perhaps creation of UiStartup and UiMain can be shown with constructors

thaddyyz replied:
![image](https://user-images.githubusercontent.com/76462596/139192467-8d700f97-dca1-4529-a4a3-b388f1e937f1.png)
void return should be dotted

LimJiaJing replied:
Should there be entities at the bottom of the sequence diagram?

adricpjw commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+3. `mainState()` will create a new `:UiMain` and calls a method `run()` on itself
+4. `run()` returns a boolean value equivalent to whether the recursive call should be made
+5. If the boolean value is true, `handleState()` would recursively call itself
+6. If the boolean value is false, `handleState()` would pass the call back to `:HappyBit`
```
Perhaps a simple loop sequence diagram and somewhat represent the recursive calls here?

adricpjw commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+When the `runCommand` function is executed for the `DeleteGoalCommand` object, the following steps as indicated by the
+sequence diagram below is carried out:
+
+![](Diagram Images/DeleteGoalCommandSequenceDiagram.png)
```
Perhaps instead of writing the full command syntax, you could omit the datatype of parameter if it is self explanatory. For example, "printManager" being PrintManager is self-explanatory.

Also, perhaps the goal.getDescription() should have been a call to the Goal object first?
![image](https://user-images.githubusercontent.com/60023580/138999268-d088672a-9cda-480c-b3ae-9133e8a354eb.png)

thaddyyz replied:
![image](https://user-images.githubusercontent.com/76462596/139193362-a6e0a3b9-4451-452b-9fbe-946505857420.png)
Possible to group the objects into UI, Logic... for clarity
Also might be better to keep labels consistent if you intend to use it

xkisxk replied:
Perhaps getGoal should be a self-invocation rather than a constructor? Because later on under DoneHabitCommand, getGoal is shown as a self-invocation. Also getGoal does not seem like a constructor name?

antrikshv commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+When the `runCommand` function is executed for the `ListHabitsCommand` object, the following steps as indicated by the
+sequence diagram below is carried out:
+
+![](Diagram Images/ListHabitsCommandSequenceDiagram.png)
```
Perhaps the Diagrams get a bit repetitive after awhile as they are very similar in nature. Maybe could consider, showing the overall logic of the command classes in an example

adricpjw commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+* `Import` will then populate `GoalList` with `Goal` and `Habit` objects accordingly
+before returning `GoalList` back to `Storage` and back to user.
+
+`Storage` class can also export data to storage file with `Export` class.
```
perhaps another diagram of visualisation can be used to elaborate on the exporting (be it Habit or Goal)?

antrikshv commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+* [Appendix C: Non-Functional Requirements](#appendix-c-non-functional-requirements)
+* [Appendix D: Glossary](#appendix-d-glossary)
+* [Appendix E: Instructions for Manual Testing](#appendix-e-instructions-for-manual-testing)
+
```
Great use of hyperlinks, and overall very well structured DG!

adricpjw reviewed (commented):
Good work! Very nice use of UML diagrams which from what I can tell error-free! Perhaps readability can be abit improved on the diagrams since some methods have many parameters, but otherwise good job

antrikshv commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+When the `runCommand` function is executed for the `AddGoalCommand` object, the following steps as indicated by the
+sequence diagram below is carried out:
+
+![](https://www.planttext.com/api/plantuml/img/ZLB1QiCm3BtxAqGltI1fxcQCZWvwwM2mwouY5fuXbZDRsVVFIGepn2sxoUz9xptBFYR1A9CVrFvBP4owwyO1UKOEVV1Tek-9kAVMEBGHlMgVOQVXnPXpmE4Kl4Ss6OWJNmyFDXCNbqJ3-LeryCbZT2nlo6WfQdWlJWqa2J5N6ZxMub5XB-u7XIfUIcqnc5DjVNCZherBg9Leu7QKqhWYbwqhw69-MtC7UdNCcUalpC6Il5Bgenl51PxldfjicU2EPZt8KzlUdpBqF_NQYXVnsb9AqHg_36wViHpRiaTYa__WBm00)
```
![image](https://user-images.githubusercontent.com/70212679/139000704-fe6a2613-fc4f-4ff0-9129-a3f8f71aa92c.png)

Perhaps it would be good to have a brief explanation as to what is going on within the diagram, as it may get a bit confusing. Can consider doing this for the other commands as well!

LimJiaJing replied:
Perhaps goalList can be renamed to goals to abide by the Java coding conventions

antrikshv commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+The class diagram shows the interactions between the different classes.
+
+![Storage Class Diagram](Diagram%20Images/StorageClassDiagram.png)
+
```
![image](https://user-images.githubusercontent.com/70212679/139001193-5ff25e3f-2aad-4516-bf9d-d8e6d9ff23d9.png)
Maybe can consider adding small notes on the arrows, to describe the interaction between the different classes, to make it more understandable and digestible for the reader

antrikshv reviewed (commented):
Overall Super well structured and comprehensive DG!

ashrafjfr commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+
+The diagram below shows the higher-level implementation of the interactions of all the components within `HappyBit` 
+
+![Ui Diagram](Diagram Images/SystemArchitecture.png)
```
On the team's DeveloperGuide.md, image is not showing. Should be an issue with the syntax for "Diagram Images".

ashrafjfr commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+The Sequence Diagram below illustrates the interactions within the `Ui` component when `run()` is called in the main
+class `HappyBit.java`
+
+![Ui Diagram](Diagram Images/UiDiagramCopy.png)
```
Image is not showing here as well. Perhaps similar syntax issue as stated above. Similar to images below that are taken from Diagram Images folder.

ashrafjfr commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+sequence diagram below is carried out:
 
-{Describe the target user profile}
+![](Diagram Images/DoneCommandSequenceDiagram.png)
```
The printManager from AddGoalCommand has a dashed line returning to the activation bar that calls for it. However, some of the other diagrams do not have it. Perhaps can keep the diagrams consistent with regards to whether the return line is displayed after printManager.

![image](https://user-images.githubusercontent.com/69456367/139170631-67a98332-5387-49c7-8888-779da8f72bb5.png)

ashrafjfr commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+
+#### `ListGoalCommand`
+
+When the `runCommand` function is executed for the `ListGoalsCommand` object, the following steps as indicated by the
```
Agree with @antrikshv comment above. The current comments for each object is too vague. More information can be shared with regards to each step being carried out so developers can follow the diagram better rather than just mentioning the following steps as indicated by the diagram below.

ashrafjfr reviewed (commented):
Overall, good attempt with the DG! Do include more information for each diagram such what the following steps are rather than just leaving the reader/developer to follow the diagram on his/her own. Also do watch your syntax as most images are not showing on the DeveloperGuide.md in your team repo.

thaddyyz commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+* Developer team: find the team [here](https://se-education.org/addressbook-level3/AboutUs.html)
+* Link to Developer Guide: <https://se-education.org/addressbook-level3/DeveloperGuide.html>
+* Link to User Guide: <https://se-education.org/addressbook-level3/UserGuide.html#quick-start>
+
```
Clear navigation tools for reference.

thaddyyz commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+
+#### `UpdateGoalNameCommand`
+
+#### `DeleteGoalCommand`
```
![image](https://user-images.githubusercontent.com/76462596/139192915-94cc0b22-8a49-4c41-9df9-cd7468157a31.png)
If the call sequence is similar can consider not creating multiple sequence diagrams but 1 and specify the difference in the comments with smaller diagrams showing the slight difference in sequence if any.

LimJiaJing commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+When the `runCommand` function is executed for the `ListGoalsCommand` object, the following steps as indicated by the
+sequence diagram below is carried out:
+
+![](Diagram Images/ListGoalsCommandSequenceDiagram.png)
```
![image](https://user-images.githubusercontent.com/80034449/139195555-35e3d8e2-4cfb-476e-8514-a1d48f7fe002.png)
Should there be a [else] at the bottom half of the alt frame?

LimJiaJing commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+sequence diagram below is carried out:
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![](Diagram Images/DeleteHabitCommandSequenceDiagram.png)
```
![image](https://user-images.githubusercontent.com/80034449/139196588-91bee3eb-f36e-4c67-a36a-b3f53d2587ae.png)
Perhaps, since the return arrows do not add value to the diagram (do not give more information) is could be omitted to make the diagram less cluttered?

LimJiaJing reviewed (commented):
The overall layout of the developer guide is very neat!

xkisxk commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+sequence diagram below is carried out:
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![](Diagram Images/DeleteHabitCommandSequenceDiagram.png)
```
Is ":DeleteGoalCommand" supposed to be ":DeleteHabitCommand" since the sequence diagram is under the DeleteHabitCommand section?

xkisxk replied:
Also, is getHabitList() suppose to be a call to goal from ":GoalList".
And the activation bar for the constructor of Goal does not end. Is it more accurate to end the constructor there, then reactivating it the next time it is called?
<img width="559" alt="1" src="https://user-images.githubusercontent.com/69447105/139376074-404928da-2a44-4ed5-9af3-3a535bbf1ea4.png">
.

xkisxk commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+{Someone shall give instructions on how to do a manual product testing 
+e.g., how to load sample data to be used for testing}
```
I think it is pretty good overall. Good use of diagrams to explain certain processes, good use of hyperlinks and everything is broken down into clear sections.

xkisxk commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,251 @@
+* `DoneParser` - Handles the parsing of inputs starting with `done`
+* `ListParser` - Handles the parsing of inputs starting with `view`
+* `SetParser` - Handles the parsing of inputs starting with `goal`
+
```
Maybe it would be better to also include a class diagram to show how the parser classes are related to each other and its relationship with the other parts of the app?

kahhe committed:
Merge pull request #172 from AY2122S1-CS2113T-F14-1/branch-StorageTest

Branch storage test

kahhe committed:
Allow for exit from startup page

kahhe committed:
Update UpdateParser.java

kahhe committed:
Merge branch 'master' into kahhe-PeDryRunBugs

kahhe committed:
Update PrintManager.java

kahhe committed:
Merge branch 'kahhe-PeDryRunBugs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into kahhe-PeDryRunBugs

kahhe committed:
Solve merge conflict due to deleted lines

DJ-Tan committed:
Merge remote-tracking branch 'upstream/master'

STAung07 committed:
Resolved merge conflict + abstracted out getIndex and getUpdateInterval to check for 0

kahhe committed:
Merge pull request #173 from AY2122S1-CS2113T-F14-1/kahhe-PeDryRunBugs

Kahhe PE dry run bugs

STAung07 committed:
Fixes #139

STAung07 committed:
Merged with kah heng bug fixes

STAung07 committed:
PE Dry Run Bugs Fixed

STAung07 committed:
PR to master

STAung07 committed:
Fixes #124

STAung07 committed:
Merge pull request #174 from AY2122S1-CS2113T-F14-1/sta-BugFixes

PE Dry Run Bug Fixes

kahhe committed:
Add test for parseUpdateGoalEndDateCommand and parseUpdateGoalTypeCommand

DJ-Tan committed:
Merge remote-tracking branch 'upstream/master'

flerovious committed:
Update Parser

flerovious committed:
Update ParserTest

flerovious committed:
Merge branch 'master' into update-parser

# Conflicts:
#	src/main/java/happybit/parser/AddParser.java
#	src/main/java/happybit/parser/DeleteParser.java
#	src/main/java/happybit/parser/DoneParser.java
#	src/main/java/happybit/parser/ListHabitParser.java
#	src/main/java/happybit/parser/Parser.java
#	src/main/java/happybit/parser/UpdateParser.java

kahhe committed:
Add Junit Tests for UpdateParser

flerovious committed:
Fix merge bugs

kahhe committed:
Fix checkStyle

kahhe committed:
Change DeleteCommand and UpdateCommand to abstract

as per Code Review

kahhe committed:
Add test for exceeding interval limit in UpdateHabitIntervalCommand

kahhe committed:
Merge branch 'master' into kahhe-TestingAndCodeReview

flerovious committed:
Fix failing parser test cases

kahhe committed:
Fix failing junit test

kahhe committed:
Update habits.txt

kahhe committed:
Update kahhe.md

kahhe committed:
Update kahhe.md

huien77 committed:
update huien's PPP

kahhe committed:
Add Index for UG Sections

kahhe committed:
Update UserGuide.md

kahhe committed:
Change PHOTO

kahhe committed:
Add FAQs for UG

STAung07 committed:
Merge pull request #176 from AY2122S1-CS2113T-F14-1/update-parser

Update parser

STAung07 committed:
Merge with Varun

STAung07 committed:
Merge with master

STAung07 committed:
Merge branch 'master' of github.com:AY2122S1-CS2113T-F14-1/tp

STAung07 committed:
Merge branch 'master' into sta-BugFixes

huien77 committed:
updated user guide

huien77 committed:
import sequence diagram and DG

huien77 committed:
export sequence diagram and DG

kahhe committed:
Merge branch 'master' into kahhe-TestingAndCodeReview

kahhe committed:
Fix error

kahhe committed:
Merge pull request #175 from AY2122S1-CS2113T-F14-1/kahhe-TestingAndCodeReview

Kahhe testing + code review + PPP

STAung07 committed:
merge with upstream

STAung07 committed:
Merge branch 'master' of github.com:AY2122S1-CS2113T-F14-1/tp

STAung07 committed:
Merge branch 'master' into sta-BugFixes

kahhe committed:
Fix typo and error

kahhe committed:
Create HappyBit_Visual_Explain.png

STAung07 committed:
Parsing of update and command in one line + Class Creation

STAung07 committed:
Fixes #90, Fixes #118

kahhe committed:
Merge pull request #178 from AY2122S1-CS2113T-F14-1/sta-BugFixes

Multiple Updates for Goal and Habit

DJ-Tan committed:
Merge remote-tracking branch 'upstream/branch-Docs'

flerovious committed:
Add testing for MainParser

STAung07 committed:
Touch Up PPP

DJ-Tan committed:
Update Overall Architecture

Added a new overall architecture and an example sequence diagram

kahhe committed:
Update UG and DG

kahhe committed:
Update code documentation and improve code quality

huien77 committed:
fixed bugs with live export of habit

huien77 committed:
Merge branch 'master' into branch-Storage

kahhe committed:
Merge pull request #181 from AY2122S1-CS2113T-F14-1/branch-Storage

Branch storage

kahhe committed:
Add documentation

huien77 committed:
update manual testing 1

huien77 committed:
Merge branch 'master' into branch-Docs

huien77 committed:
update manual testing 2

added launching and exiting the program

STAung07 committed:
Merge with Master and Huien DG branch

STAung07 committed:
Merge with master

STAung07 committed:
Merge branch 'branch-Docs' of github.com:AY2122S1-CS2113T-F14-1/tp into branch-Docs

kahhe committed:
Add more accurate error message

kahhe committed:
Add code documentation

huien77 committed:
update manual testing 3

added starting application

kahhe committed:
Improve accuracy of error message

kahhe committed:
Merge branch 'master' into sta-BugFixes

kahhe committed:
Merge pull request #180 from AY2122S1-CS2113T-F14-1/branch-UG

Branch UG

DJ-Tan committed:
Update UI Component

DJ-Tan committed:
Merge remote-tracking branch 'upstream/branch-Docs' into branch-Docs

STAung07 committed:
Merge with Daren

STAung07 committed:
resolved merge conflict

STAung07 committed:
Refactored Files

STAung07 committed:
Added Sequence Diagram for AddHabit + fixed merge bug in Implementation section

flerovious committed:
Merge pull request #179 from AY2122S1-CS2113T-F14-1/update-MainParser

Add testing for MainParser

DJ-Tan committed:
Update Parser Component

DJ-Tan committed:
Merge remote-tracking branch 'upstream/branch-Docs' into branch-Docs

DJ-Tan committed:
Fix Merge Conflict

Fix Merge Conflict in Parser Component

flerovious committed:
Update DG formatting

kahhe committed:
Explain repeated parameters in UG

DJ-Tan committed:
Update Command, Goal, Storage Component

Included a class diagram for command, goal and storage components.

DJ-Tan committed:
Merge remote-tracking branch 'upstream/branch-Docs' into branch-Docs

DJ-Tan committed:
Update Ui Component diagrams

Updated diagrams as per Varun's specifications

STAung07 committed:
Added Add Parser PUML for AddHabit and AddGoal Command

STAung07 committed:
Merge branch 'branch-Docs' of github.com:AY2122S1-CS2113T-F14-1/tp into branch-Docs

huien77 committed:
update manual testing 4

added section for testing deletion of goals and/or habits

kahhe committed:
Improve error message accuracy

kahhe committed:
Merge branch 'master' into sta-BugFixes

kahhe committed:
Fix error

huien77 committed:
update manual testing 5

added erroneous test cases

STAung07 committed:
Merge pull request #182 from AY2122S1-CS2113T-F14-1/sta-BugFixes

Documentation and error messages

kahhe committed:
Update UserGuide.md

kahhe committed:
Add visual explanation

STAung07 committed:
List and Done Command Sequence Diagrams + Explanation

STAung07 committed:
Merge branch 'branch-Docs' of github.com:AY2122S1-CS2113T-F14-1/tp into branch-Docs

kahhe committed:
Update DeveloperGuide.md

STAung07 committed:
Merge branch 'branch-Docs' of github.com:AY2122S1-CS2113T-F14-1/tp into branch-Docs

huien77 committed:
update storage to clear data file if corrupted

huien77 committed:
fixed Storage

DJ-Tan committed:
Update Adding a Goal Feature

DJ-Tan committed:
Merge remote-tracking branch 'upstream/branch-Docs' into branch-Docs

huien77 committed:
update manual testing 6

section to test storage component

DJ-Tan committed:
Merge remote-tracking branch 'upstream/master'

DJ-Tan committed:
Merge remote-tracking branch 'upstream/master' into branch-Docs

huien77 committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

STAung07 committed:
Merge with master

STAung07 committed:
Merge branch 'branch-Docs' of github.com:AY2122S1-CS2113T-F14-1/tp into branch-Docs

kahhe committed:
Update HappyBit_Visual_Explain.png

kahhe committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

STAung07 committed:
rename to ParserManager

flerovious committed:
Refactor ParserManager

flerovious committed:
Merge remote-tracking branch 'origin/branch-Docs' into branch-Docs

DJ-Tan committed:
Update ParserComponent Diagram

Updated diagram changing MainParser to ParserManager

STAung07 committed:
Merge branch 'branch-Docs' of github.com:AY2122S1-CS2113T-F14-1/tp into branch-Docs

DJ-Tan committed:
Merge remote-tracking branch 'upstream/branch-Docs' into branch-Docs

STAung07 committed:
Merge branch 'branch-Docs' of github.com:AY2122S1-CS2113T-F14-1/tp into branch-Docs

huien77 committed:
update manual testing 7

huien77 committed:
update images

huien77 committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

DJ-Tan committed:
Delete duplicate directory

DJ-Tan committed:
Merge remote-tracking branch 'upstream/branch-Docs' into branch-Docs

kahhe committed:
Change name of folder with spaces to underscore

kahhe committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

STAung07 committed:
UpdateGoalCommandParserSequenceDiagram

STAung07 committed:
Merge branch 'branch-Docs' of github.com:AY2122S1-CS2113T-F14-1/tp into branch-Docs

huien77 committed:
updated images

huien77 committed:
updating manual testing 8

added corrupted file section

huien77 committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

huien77 committed:
update huien PPP

kahhe committed:
Update DeleteCommand puml

STAung07 committed:
Finished Diagrams for Update Goal and Update Habit

STAung07 committed:
Merge branch 'branch-Docs' of github.com:AY2122S1-CS2113T-F14-1/tp into branch-Docs

STAung07 committed:
fixed bugs in diagram

kahhe committed:
Update DeveloperGuide.md

kahhe committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

kahhe committed:
Update DeleteHabitCommand Diagram

kahhe committed:
Merge pull request #177 from AY2122S1-CS2113T-F14-1/branch-Docs

Branch docs

kahhe committed:
Update HappyBit_Visual_Explain.png

kahhe committed:
Update UserGuide.md

DJ-Tan committed:
Update Adding a Habit Feature

DJ-Tan committed:
Update Listing all Goals Feature

kahhe committed:
Add UpdateGoalCommand test

flerovious committed:
Add DG section 4.4.1.

kahhe committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

kahhe committed:
Fix checkstyle

flerovious committed:
Add DG section 4.5.1.

STAung07 committed:
fixed sequence diagram bugs

STAung07 committed:
Merge branch 'branch-Docs' of github.com:AY2122S1-CS2113T-F14-1/tp into branch-Docs

flerovious committed:
Add AddGoalCommandSequenceDiagram.png

flerovious committed:
Merge remote-tracking branch 'origin/branch-Docs' into branch-Docs

kahhe committed:
Improve DG user-centeredness

kahhe committed:
Update DeleteGoalCommand Diagram

kahhe committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

STAung07 committed:
Update UG and PPP

STAung07 committed:
Merge branch 'branch-Docs' of github.com:AY2122S1-CS2113T-F14-1/tp into branch-Docs

kahhe committed:
Update DeleteCommand puml

kahhe committed:
Add DeletePraser sequence diagrams

STAung07 committed:
Added colour to all sequence diagrams

STAung07 committed:
Fixed merge conflicts

STAung07 committed:
Colour for Delete Parser

flerovious committed:
Refactor UpdateParser

kahhe committed:
Update ListGoals sequence diagrams

kahhe committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

flerovious committed:
Add DG Section 4.6.1.

flerovious committed:
Merge remote-tracking branch 'origin/branch-Docs' into branch-Docs

kahhe committed:
Update ListGoalParserCommandSequenceDiagram

kahhe committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

flerovious committed:
Remove unused classes

kahhe committed:
Update DG List Goal Implementation

DJ-Tan committed:
Fix date issue in AddParser

Fix and issue where a date longer than 8 characters is still treated as a valid date

DJ-Tan committed:
Merge remote-tracking branch 'upstream/branch-Docs' into branch-Docs

kahhe committed:
Resolve merge conflict

kahhe committed:
Update TOC

DJ-Tan committed:
Remove commented out code

DJ-Tan committed:
Merge remote-tracking branch 'upstream/branch-Docs' into branch-Docs

flerovious committed:
Add DG Section 4.7.1.

kahhe committed:
Update PPP

kahhe committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

flerovious committed:
Update DG Section 4.10.1.

flerovious committed:
Update DG Section 4.8.1.

flerovious committed:
Update DG Section 4.9.1.

kahhe committed:
Add DG import data

kahhe committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

DJ-Tan committed:
Update DG Section 4.4.2.

DJ-Tan committed:
Merge remote-tracking branch 'upstream/branch-Docs' into branch-Docs

kahhe committed:
Add DG 4.11.1

kahhe committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

DJ-Tan committed:
Update DG Section 4.5.2.

DJ-Tan committed:
Merge remote-tracking branch 'upstream/branch-Docs' into branch-Docs

DJ-Tan committed:
Update DG Section 4.6.2.

DJ-Tan committed:
Update DG Section 4.7.2.

huien77 committed:
update user guide

huien77 committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

DJ-Tan committed:
Update DG Section 4.8.2.

DJ-Tan committed:
Merge remote-tracking branch 'upstream/branch-Docs' into branch-Docs

flerovious committed:
Update PPP

kahhe committed:
Fix DeleteHabitCommandSequenceDiagram

kahhe committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

DJ-Tan committed:
Update DG Section 4.9/10/11.2.

DJ-Tan committed:
Merge remote-tracking branch 'upstream/branch-Docs' into branch-Docs

kahhe committed:
Update kahhe.md

kahhe committed:
Merge branch 'branch-Docs' of https://github.com/AY2122S1-CS2113T-F14-1/tp into branch-Docs

DJ-Tan committed:
Update sequence diagrams for Update Command

Change int array to ArrayList<Boolean>

DJ-Tan committed:
Merge remote-tracking branch 'upstream/branch-Docs' into branch-Docs

flerovious committed:
Update DoneParserTest

flerovious committed:
Remove InputParserTest.java

kahhe committed:
Fix minor errors

flerovious committed:
UpdateParserTest

DJ-Tan committed:
Merge branch 'branch-Docs'

kahhe committed:
Fix test error

DJ-Tan committed:
Merge pull request #183 from AY2122S1-CS2113T-F14-1/branch-Docs

Final changes and refinement

DJ-Tan committed:
Merge remote-tracking branch 'upstream/master'

flerovious committed:
Fix bugs in UpdateParserTest

DJ-Tan committed:
Page Break Testing

flerovious committed:
Fix checkStyleTest error

kahhe committed:
Merge pull request #184 from AY2122S1-CS2113T-F14-1/branch-Docs

FINAL

DJ-Tan committed:
Page Break Test v1.1

DJ-Tan committed:
DG Page Break Test v1.1

DJ-Tan committed:
Merge remote-tracking branch 'origin/master'

STAung07 committed:
User Guide Page Breaks

STAung07 committed:
Merge branch 'branch-Docs' of github.com:AY2122S1-CS2113T-F14-1/tp into branch-Docs

flerovious committed:
Update directory name

flerovious committed:
Merge pull request #185 from AY2122S1-CS2113T-F14-1/update-directory-name

Update directory name

DJ-Tan committed:
DG Page Break Test v1.2

STAung07 committed:
Updated PPP

DJ-Tan committed:
DG Page Break Test v1.3

STAung07 committed:
Merge pull request #186 from AY2122S1-CS2113T-F14-1/branch-Docs

Branch docs

flerovious committed:
Fix PPP broken link

DJ-Tan committed:
Merge remote-tracking branch 'upstream/master'

kahhe committed:
fix link PPP

DJ-Tan committed:
Merge remote-tracking branch 'upstream/fix-PPP-bug'

flerovious committed:
Merge pull request #187 from AY2122S1-CS2113T-F14-1/fix-PPP-bug

Fix PPP broken link

DJ-Tan committed:
Update JAR file link

DJ-Tan committed:
Merge remote-tracking branch 'upstream/master'

STAung07 committed:
Update User Guide for CS2101 submission

STAung07 committed:
added link to demo video

STAung07 committed:
Grammar fixes

DJ-Tan committed:
Update User Guide examples

DJ-Tan committed:
Merge remote-tracking branch 'upstream/branch-Docs' into branch-Docs

DJ-Tan committed:
Fix Checkstyle issues

Fix issues with checkstyle after merge
